packages/car/src/index.ts

```diff
@@ -189,6 +194,14 @@ class DefaultCar implements Car {
     for (const root of roots) {
       void queue.add(async () => {
         await this.#walkDag(root, queue, async (cid, bytes) => {
+          // check if duplicate blocks should be skipped
+          if (typeof options?.blockFilter !== 'undefined') {
+            // skip blocks that have already been written
+            if (options?.blockFilter.has(cid.toString())) {
+              return
+            }
+            options?.blockFilter.add(cid.toString())
```

Comment:
`options.blockFilter` is guaranteed to be type `Filter` in this branch

```suggestion
            options.blockFilter.add(cid.toString())
```